File: DemoServer/PythonDemo/demo_methods/Basic/Delete.py
from demo.holder import DocumentStoreHolder
from demo.entities import Company


class Delete(object):
    @staticmethod
    def delete(company_id=2):
        with DocumentStoreHolder.get_store().open_session() as session:
            session.delete(session.load("companies/" + company_id, object_type=Company))
            session.save_changes()
            return "CompanyId {0} Deleted  Successfully.".format(company_id)

File: DemoServer/JavaDemo/src/main/java/net/ravendb/demo/basic/Delete.java
package net.ravendb.demo.basic;

import net.ravendb.client.documents.session.IDocumentSession;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.entities.Company;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class Delete {

    @GetMapping("/basic/delete")
    public String delete(
            @RequestParam (value = "companyId", defaultValue = "companies/2-A") String companyId) {
        try (IDocumentSession session = DocumentStoreHolder.getStore().openSession()) {
            session.delete(session.load(Company.class, companyId));
            session.saveChanges();

            return String.format("%s Deleted Successfully", companyId);
        }
    }
}
